### ISSUE TITLE
Update always fails in Debian

### ISSUE THREAD

**Issue by datwaft** [wbthomason/packer.nvim#95]
Hello, after migrating my vim configuration from Ubuntu to Debian I am currently having this problem: every time I use the `sync` or `update` command it fails for every plugin (but it works for the `install` command). The problem didn't happen with the same configuration in Ubuntu.

Here is a screenshot with only _packer_ as a plugin (after commenting my configuration file and plugins file): 
![image](https://user-images.githubusercontent.com/37723586/99869941-a6e34b00-2b94-11eb-92e6-0553986a0641.png)

And here is a screenshot reinstalling my plugins after uncommenting the comments:
![image](https://user-images.githubusercontent.com/37723586/99870001-2f61eb80-2b95-11eb-915f-1b4f27f3d962.png)

And here a screen of the `sync` command after that:
![image](https://user-images.githubusercontent.com/37723586/99870025-6c2de280-2b95-11eb-863d-8ef1ef6530f9.png)

~I have my suspicions that it may work after some plugin updates and there is a difference, but I would need to wait to know.~ It doesn't work after there is an update.

Here is my `packer_compiled.vim`: [packer_compiled.zip](https://github.com/wbthomason/packer.nvim/files/5577168/packer_compiled.zip)

<details>
<summary>packer_compiled.vim</summary>

```viml
" Automatically generated packer.nvim plugin loader code

if !has('nvim')
  finish
endif

lua << END
local plugins = {
  ["packer.nvim"] = {
    loaded = false,
    only_sequence = false,
    only_setup = false,
    path = "/home/datwaft/.local/share/nvim/site/pack/packer/opt/packer.nvim"
  }
}

local function handle_bufread(names)
  for _, name in ipairs(names) do
    local path = plugins[name].path
    for _, dir in ipairs({ 'ftdetect', 'ftplugin', 'after/ftdetect', 'after/ftplugin' }) do
      if #vim.fn.finddir(dir, path) > 0 then
        vim.cmd('doautocmd BufRead')
        return
      end
    end
  end
end

_packer_load = nil

local function handle_after(name, before)
  local plugin = plugins[name]
  plugin.load_after[before] = nil
  if next(plugin.load_after) == nil then
    _packer_load({name}, {})
  end
end

_packer_load = function(names, cause)
  local some_unloaded = false
  for _, name in ipairs(names) do
    if not plugins[name].loaded then
      some_unloaded = true
      break
    end
  end

  if not some_unloaded then return end

  local fmt = string.format
  local del_cmds = {}
  local del_maps = {}
  for _, name in ipairs(names) do
    if plugins[name].commands then
      for _, cmd in ipairs(plugins[name].commands) do
        del_cmds[cmd] = true
      end
    end

    if plugins[name].keys then
      for _, key in ipairs(plugins[name].keys) do
        del_maps[key] = true
      end
    end
  end

  for cmd, _ in pairs(del_cmds) do
    vim.cmd('silent! delcommand ' .. cmd)
  end

  for key, _ in pairs(del_maps) do
    vim.cmd(fmt('silent! %sunmap %s', key[1], key[2]))
  end

  for _, name in ipairs(names) do
    if not plugins[name].loaded then
      vim.cmd('packadd ' .. name)
      if plugins[name].config then
        for _i, config_line in ipairs(plugins[name].config) do
          loadstring(config_line)()
        end
      end

      if plugins[name].after then
        for _, after_name in ipairs(plugins[name].after) do
          handle_after(after_name, name)
          vim.cmd('redraw')
        end
      end

      plugins[name].loaded = true
    end
  end

  handle_bufread(names)

  if cause.cmd then
    local lines = cause.l1 == cause.l2 and '' or (cause.l1 .. ',' .. cause.l2)
    vim.cmd(fmt('%s%s%s %s', lines, cause.cmd, cause.bang, cause.args))
  elseif cause.keys then
    local keys = cause.keys
    local extra = ''
    while true do
      local c = vim.fn.getchar(0)
      if c == 0 then break end
      extra = extra .. vim.fn.nr2char(c)
    end

    if cause.prefix then
      local prefix = vim.v.count and vim.v.count or ''
      prefix = prefix .. '"' .. vim.v.register .. cause.prefix
      if vim.fn.mode('full') == 'no' then
        if vim.v.operator == 'c' then
          prefix = '' .. prefix
        end

        prefix = prefix .. vim.v.operator
      end

      vim.fn.feedkeys(prefix, 'n')
    end

    -- NOTE: I'm not sure if the below substitution is correct; it might correspond to the literal
    -- characters \<Plug> rather than the special <Plug> key.
    vim.fn.feedkeys(string.gsub(string.gsub(cause.keys, '^<Plug>', '\\<Plug>') .. extra, '<[cC][rR]>', '\r'))
  elseif cause.event then
    vim.cmd(fmt('doautocmd <nomodeline> %s', cause.event))
  elseif cause.ft then
    vim.cmd(fmt('doautocmd <nomodeline> %s FileType %s', 'filetypeplugin', cause.ft))
    vim.cmd(fmt('doautocmd <nomodeline> %s FileType %s', 'filetypeindent', cause.ft))
  end
end

-- Runtimepath customization

-- Pre-load configuration
-- Post-load configuration
-- Config for: vim-tmux-navigator
loadstring("\27LJ\2\2D\0\0\2\0\3\0\0056\0\0\0009\0\1\0)\1\1\0=\1\2\0K\0\1\0'tmux_navigator_disable_when_zoomed\6g\bvim\0")()
-- Config for: nvim-treesitter
loadstring("\27LJ\2\2ø\2\0\0\5\0\19\0\0236\0\0\0'\1\1\0B\0\2\0029\0\2\0005\1\3\0005\2\4\0004\3\0\0=\3\5\2=\2\6\0015\2\a\0=\2\b\0015\2\n\0005\3\t\0=\3\v\0025\3\f\0=\3\r\0025\3\14\0005\4\15\0=\4\16\3=\3\17\2=\2\18\1B\0\2\1K\0\1\0\rrefactor\15navigation\fkeymaps\1\0\2\20goto_next_usage\n<a-*>\24goto_previous_usage\n<a-#>\1\0\1\venable\2\28highlight_current_scope\1\0\1\venable\1\26highlight_definitions\1\0\0\1\0\1\venable\2\vindent\1\0\1\venable\1\14highlight\fdisable\1\0\1\venable\2\1\0\1\21ensure_installed\15maintained\nsetup\28nvim-treesitter.configs\frequire\0")()
-- Config for: vim-slime
loadstring("\27LJ\2\2å\1\0\0\3\0\v\0\0186\0\0\0009\0\1\0'\1\3\0=\1\2\0006\0\0\0009\0\1\0)\1\1\0=\1\4\0006\0\5\0009\0\6\0'\1\a\0'\2\b\0B\0\3\0016\0\0\0009\0\1\0005\1\n\0=\1\t\0K\0\1\0\1\0\2\16socket_name\fdefault\16target_pane\v{last}\25slime_default_config\26<Plug>SlimeMotionSend\14<leader>s\tnmap\tvimp\25slime_python_ipython\ttmux\17slime_target\6g\bvim\0")()
-- Config for: vim-nightfly-guicolors
loadstring("\27LJ\2\2\1\0\0\2\0\a\0\r6\0\0\0009\0\1\0+\1\2\0=\1\2\0006\0\0\0009\0\3\0)\1\1\0=\1\4\0006\0\0\0009\0\5\0'\1\6\0B\0\2\1K\0\1\0\27 colorscheme nightfly \bcmd\24nightflyCursorColor\6g\18termguicolors\6o\bvim\0")()
-- Config for: camelcasemotion
loadstring("\27LJ\2\2>\0\0\2\0\4\0\0056\0\0\0009\0\1\0'\1\3\0=\1\2\0K\0\1\0\r<leader>\24camelcasemotion_key\6g\bvim\0")()
-- Config for: quick-scope
loadstring("\27LJ\2\2B\0\0\2\0\4\0\0056\0\0\0009\0\1\0005\1\3\0=\1\2\0K\0\1\0\1\5\0\0\6f\6F\6t\6T\25qs_highlight_on_keys\6g\bvim\0")()
-- Config for: lightline.vim
loadstring("\27LJ\2\2\3\0\0\5\0\r\0\0196\0\0\0009\0\1\0'\1\2\0B\0\2\0016\0\0\0009\0\3\0005\1\5\0005\2\b\0004\3\3\0005\4\6\0>\4\1\0035\4\a\0>\4\2\3=\3\t\2=\2\n\0015\2\v\0=\2\f\1=\1\4\0K\0\1\0\23component_function\1\0\2\20currentfunction\23CocCurrentFunction\14cocstatus\15coc#status\vactive\tleft\1\0\0\1\6\0\0\14cocstatus\20currentfunction\rreadonly\rfilename\rmodified\1\3\0\0\tmode\npaste\1\0\1\16colorscheme\rnightfly\14lightline\6gs      function! CocCurrentFunction()\n         return get(b:, 'coc_current_function', '')\n      endfunction\n   \bcmd\bvim\0")()
-- Config for: coc.nvim
loadstring("\27LJ\2\2½\1\0\0\3\0\n\0\0226\0\0\0009\0\1\0009\0\2\0005\1\3\0006\2\0\0009\2\4\0029\2\5\2B\0\3\2)\1\0\0\3\1\0\0X\0\56\0\0\0009\0\6\0'\1\a\0B\0\2\1X\0\56\0\0\0009\0\1\0009\0\b\0'\1\t\0B\0\2\1K\0\1\0\fdoHover\14CocAction% execute 'h '.expand('<cword>') \bcmd\rfiletype\abo\1\3\0\0\bvim\nindex\nindex\afn\bvim7\0\0\2\0\4\0\0066\0\0\0009\0\1\0009\0\2\0'\1\3\0B\0\2\1K\0\1\0\vformat\14CocAction\afn\bvim;\0\1\4\0\4\0\a6\1\0\0009\1\1\0019\1\2\1'\2\3\0\18\3\0\0B\1\3\1K\0\1\0\vformat\14CocAction\afn\bvim\\\0\0\3\0\5\0\a6\0\0\0009\0\1\0009\0\2\0'\1\3\0'\2\4\0B\0\3\1K\0\1\0!editor.action.organizeImport\15runCommand\14CocAction\afn\bvim¥\14\1\0\5\0K\0¿\0016\0\0\0009\0\1\0005\1\3\0=\1\2\0003\0\4\0006\1\0\0009\1\5\1+\2\2\0=\2\6\0016\1\0\0009\1\5\1)\2\2\0=\2\a\0016\1\0\0009\1\5\1)\2,\1=\2\b\0016\1\0\0009\1\5\0016\2\0\0009\2\5\0029\2\t\2'\3\n\0&\2\3\2=\2\t\1'\1\v\0006\2\0\0009\2\5\2=\1\f\0026\2\0\0009\2\r\2=\1\f\0026\1\0\0009\1\14\1'\2\15\0B\1\2\0016\1\16\0009\1\17\0015\2\18\0'\3\19\0'\4\20\0B\1\4\0016\1\16\0009\1\17\0015\2\21\0'\3\22\0'\4\23\0B\1\4\0016\1\16\0009\1\17\0015\2\24\0'\3\25\0'\4\26\0B\1\4\0016\1\16\0009\1\17\0015\2\27\0'\3\28\0'\4\29\0B\1\4\0016\1\16\0009\1\17\0015\2\30\0'\3\31\0'\4 \0B\1\4\0016\1\16\0009\1\17\0015\2!\0'\3\"\0'\4#\0B\1\4\0016\1\16\0009\1$\0015\2%\0'\3&\0\18\4\0\0B\1\4\0016\1\16\0009\1\17\1'\2'\0'\3(\0B\1\3\0016\1\16\0009\1)\1'\2*\0'\3+\0B\1\3\0016\1\16\0009\1\17\1'\2*\0'\3+\0B\1\3\0016\1\16\0009\1)\1'\2,\0'\3-\0B\1\3\0016\1\16\0009\1\17\1'\2,\0'\3-\0B\1\3\0016\1\16\0009\1\17\1'\2.\0'\3/\0B\1\3\0016\1\16\0009\1\17\0015\0020\0'\0031\0'\0042\0B\1\4\0016\1\16\0009\1)\0015\0023\0'\0031\0'\0042\0B\1\4\0016\1\16\0009\1$\0015\0024\0'\0035\0'\0046\0B\1\4\0016\1\16\0009\1)\1'\0027\0'\0038\0B\1\3\0016\1\16\0009\0019\1'\0027\0'\0038\0B\1\3\0016\1\16\0009\1)\1'\2:\0'\3;\0B\1\3\0016\1\16\0009\0019\1'\2:\0'\3;\0B\1\3\0016\1\16\0009\1)\1'\2<\0'\3=\0B\1\3\0016\1\16\0009\0019\1'\2<\0'\3=\0B\1\3\0016\1\16\0009\1)\1'\2>\0'\3?\0B\1\3\0016\1\16\0009\0019\1'\2>\0'\3?\0B\1\3\0016\1\16\0009\1@\1'\2A\0003\3B\0B\1\3\0016\1\16\0009\1@\1'\2C\0003\3D\0B\1\3\0016\1\16\0009\1@\1'\2E\0003\3F\0B\1\3\0016\1\16\0009\1\17\1'\2G\0'\3H\0B\1\3\0016\1\0\0009\1\1\0015\2J\0=\2I\1K\0\1\0\1\3\0\0\bvim\thelp\30markdown_fenced_languages\29:CocCommand explorer<CR>\n<C-n>\0\aOR\0\tFold\0\vFormat\16map_command\27<Plug>(coc-classobj-a)\aac\27<Plug>(coc-classobj-i)\aic\26<Plug>(coc-funcobj-a)\aaf\tomap\26<Plug>(coc-funcobj-i)\aif\":<C-u>CocList diagnostics<cr>\14<leader>d\1\2\0\0\vsilent\1\2\0\0\vsilent\29<Plug>(coc-range-select)\n<C-s>\1\2\0\0\vsilent\29 <Plug>(coc-fix-current)\15<leader>qf% <Plug>(coc-codeaction-selected)\14<leader>a <Plug>(coc-format-selected)\14<leader>f\txmap\23<Plug>(coc-rename)\15<leader>rn\6K\1\2\0\0\vsilent\rnnoremap\27<Plug>(coc-references)\agr\1\2\0\0\vsilent\31<Plug>(coc-implementation)\agi\1\2\0\0\vsilent <Plug>(coc-type-definition)\agy\1\2\0\0\vsilent\27<Plug>(coc-definition)\agd\1\2\0\0\vsilent <Plug>(coc-diagnostic-next)\a]g\1\2\0\0\vsilent <Plug>(coc-diagnostic-prev)\a[g\1\2\0\0\vsilent\tnmap\tvimpC autocmd CursorHold * silent call CocActionAsync('highlight') \bcmd\awo\15signcolumn\byes\6c\14shortmess\15updatetime\14cmdheight\vhidden\6o\0\1\16\0\0\17coc-explorer\17coc-snippets\fcoc-lua\15coc-vimlsp\15coc-python\rcoc-json\rcoc-html\fcoc-xml\fcoc-css\14coc-emmet\17coc-tsserver\22coc-tslint-plugin\14coc-vetur\rcoc-java\15coc-clangd\26coc_global_extensions\6g\bvim\0")()
-- Conditional loads
-- Load plugins in order defined by `after`
END

function! s:load(names, cause) abort
call luaeval('_packer_load(_A[1], _A[2])', [a:names, a:cause])
endfunction


" Command lazy-loads

" Keymap lazy-loads

augroup packer_load_aucmds
  au!
  " Filetype lazy-loads
  " Event lazy-loads
augroup END
```

</details>

**Comment by wbthomason** [comment]
Thanks for the report! Could you tell me what your `git` version is (and ideally what your `git` version was on Ubuntu)?

I suspect this might be the issue hypothetically fixed by https://github.com/wbthomason/packer.nvim/pull/54; if you can confirm that this PR fixes the problem for you, I can go ahead and merge it (it's just waiting on validation from someone who can repro the problem).

**Comment by datwaft** [comment]
It seems Debian has version `2.20.1` and Ubuntu has version `2.25.1`.

Here some screenshots:

**Debian:**
![image](https://user-images.githubusercontent.com/37723586/100013031-d020f880-2d99-11eb-8006-e60ddc33bc98.png)

**Ubuntu:**
![image](https://user-images.githubusercontent.com/37723586/100013050-d7480680-2d99-11eb-8bc5-970a054725fb.png)

---

The pull request branch seems to fail when installing, but after that the packages are there and can be updated (or at least most of them) :

**Cannot be installed:**
![image](https://user-images.githubusercontent.com/37723586/100014650-4a527c80-2d9c-11eb-85f3-bbe9958872bb.png)

**Can be updated:**
![image](https://user-images.githubusercontent.com/37723586/100014765-7a9a1b00-2d9c-11eb-97d7-d1aa55f686a9.png)

But it seems they are not installed correctly because `tree-sitter` gives me some problems.

---

Here with only `vimppecable` as a plugin:

**Cannot be installed:**
![image](https://user-images.githubusercontent.com/37723586/100015031-eda39180-2d9c-11eb-849d-924cfd39ebb9.png)

**Can be updated;**
![image](https://user-images.githubusercontent.com/37723586/100015074-fc8a4400-2d9c-11eb-87a6-416212c774ad.png)

---

## Additional details

<details>
<summary>Additional details</summary>

I used these commands to use that branch:

```shell
rm -rdf ~/.local/share/nvim/site/pack/packer
git clone https://github.com/wbthomason/packer.nvim --branch="fix-old-git-compat" ~/.local/share/nvim/site/pack/packer/opt/packer.nvim
```
And these is my `packer` configuration:

```lua
vim.cmd [[ packadd packer.nvim ]]

vim.cmd [[ autocmd BufWritePost plugins.lua PackerCompile ]]

return require('packer').startup(function()
   use { 'wbthomason/packer.nvim', opt = true, branch = 'fix-old-git-compat' }
   use { 'svermeulen/vimpeccable' }
end)
```

</details>

**Comment by carlitux** [comment]
@datwaft about `vim-slime` you need to put the branch to `main` github now use main instead of master. for later also share your packer set up so you can have better help.

**Comment by datwaft** [comment]
> @datwaft about `vim-slime` you need to put the branch to `main`, github now use main instead of master. for later also share your packer set up so you can have better help.

Thanks for the tip about `vim-slime`. Curiously, that error only happens on **Debian** with that PR branch, so it is probably fixed in the lastest version of `packer`.

Also, I put my packer setup in **Additional details**.

I am currently using **Ubuntu** as my main WSL OS and keeping **Debian** for if you need more information.

**Comment by datwaft** [comment]
Additionally, every time I open nvim this error appears:

![image](https://user-images.githubusercontent.com/37723586/100026045-a672cc00-2daf-11eb-81d8-db7a009e8360.png)

**Comment by wbthomason** [comment]
Ah, @datwaft, sorry about that - the PR I requested you to try failing is entirely my fault; it was missing commits from `master`. If you have a chance, could you please try it again?

**Comment by datwaft** [comment]
I just tested it, and it seems to work perfectly.

Here are some screenshots:

<details>
<summary>Screenshots</summary>

#### Install

![image](https://user-images.githubusercontent.com/37723586/100132694-66612700-2e4b-11eb-8f04-dfd83021c51c.png)

**Note:** `nvim-treesitter-refactor` fails because I have it configured to _require_ `treesitter` or so I think, curiously `nvim-treesitter-context` also _requires_ `treesitter` but it doesn't fail.

#### Update

![image](https://user-images.githubusercontent.com/37723586/100132510-2306b880-2e4b-11eb-8155-b0ea5fa6c58e.png)


</details>

**Comment by wbthomason** [comment]
Thanks! It's strange that you do still see the one failure - does that happen repeatably?

Separate from that, I'll go ahead and merge the PR to fix this issue on `master`. Thank you for the help!

**Comment by datwaft** [comment]
> Thanks! It's strange that you do still see the one failure - does that happen repeatably?

No, it only happens the first time I do the install. I have my suspicions that it's because that plugin is configured to require `treesitter`, and `treesitter` hasn't been installed yet when I do the first install.

**Comment by wbthomason** [comment]
That shouldn't happen, though - everything should be installed at once in that case. Does it happen *every* first install?

**Comment by datwaft** [comment]
> That shouldn't happen, though - everything should be installed at once in that case. Does it happen _every_ first install?

Yes, it happens every time I remove `packer` and clone it again, as I was doing when testing that pull request.

I am going to open a new Issue for that, should I close this Issue now or do you close it?

**Comment by wbthomason** [comment]
Either way - I can close it for now. Thanks!
